internal/store/store.go

```diff
@@ -0,0 +1,68 @@
+package store
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/metal-toolbox/bioscfg/internal/configuration"
+	"github.com/metal-toolbox/bioscfg/internal/model"
+	"github.com/metal-toolbox/bioscfg/internal/store/fleetdb"
+	"github.com/pkg/errors"
+)
+
+type Repository interface {
+	// AssetByID returns asset based on the identifier.
+	AssetByID(ctx context.Context, assetID uuid.UUID) (*model.Asset, error)
+}
+
+func NewRepository(ctx context.Context, config *configuration.Configuration) (Repository, error) {
+	switch config.StoreKind {
+	case model.FleetDB:
+		return fleetdb.New(ctx, config.FleetDBOptions)
+	case model.MemDB:
```

Comment:
If we're really going to include something like a `MemDB` for testing (presumably) we should bundle all the required stuff together under a `TestMode` flag. I feel like we're going to carry a lot of work with us supporting `MemDB` when in fact outside of actually bootstrapping the service during development we're nevert going to use that mode.

Reply:
I was using it during my early development so I didn't have to worry about connecting to fleetdb. But it's not really in use anymore, and a mock would be better for unit testing. So I just removed it.